Tasks/PowerShellV2/powershell.ts

```diff
@@ -45,13 +46,28 @@ async function run() {
         console.log(tl.loc('GeneratingScript'));
         let contents: string[] = [];
         contents.push(`$ErrorActionPreference = '${input_errorActionPreference}'`);
-        if (input_targetType.toUpperCase() == 'FILEPATH') {
-            contents.push(`. '${input_filePath.replace(/'/g, "''")}' ${input_arguments}`.trim());
-            console.log(tl.loc('JS_FormattedCommand', contents[contents.length - 1]));
-        }
-        else {
-            contents.push(input_script);
+        let script = '';
+        if (input_targetType.toUpperCase() == 'FILEPATH') {            
+            script = `. '${input_filePath.replace(/'/g, "''")}' ${input_arguments}`.trim();
+        }else{
+            script = `${input_script}`;
+        }        
+        if(input_showWarnings){
+            script = `
+                $warnings = New-Object System.Collections.ObjectModel.ObservableCollection[System.Management.Automation.WarningRecord];
+                Register-ObjectEvent -InputObject $warnings -EventName CollectionChanged -Action {
+                    if($Event.SourceEventArgs.Action -like "Add"){
+                        $Event.SourceEventArgs.NewItems | ForEach-Object {
+                            Write-Host "##vso[task.logissue type=warning;]$_";
+                        }
+                    }
+                };
+                Invoke-Command {${script}} -WarningVariable +warnings;
+            `;
         }
+        contents.push(script);
+        // log with detail to avoid a warning output.
+        tl.logDetail(uuidV4(),tl.loc('JS_FormattedCommand', script),null, 'command', 'command', 0);
```

Comment:
I see you're using `tl.logDetail` here where the rest of this is using `console.log`. Perhpas for consistency this should remain the same or (and I suspect this is the case) if you only want this for a certain log level the `console.log` should most likely be replaced.

Reply:
No I can't use console.log. See the comment above. This is the script that contains `##vso[task.logissue type=warning;]`. This will trigger always a warning. So to prevent this, I force with tl.logDetail that it isn't important.

Reply:
@sandermvanvliet  so is it okay, like it is? Or is there a good method to escape `##vso[task.logissue type=warning;]`?